Added standard ball launch

TTibz27 opened the pull request:
Added a ball launcher object, made the baseball a prefab, set up a debug function to launch the ball, messed around with timers to destroy prefabs and to scale ball/ disable hitboxes while its starting up.

TTibz27 committed:
Added standard ball launch

Added a ball launcher object, made the baseball a prefab, set up a debug function to launch the ball, messed around with timers to destroy prefabs and to scale ball/ disable hitboxes while its starting up.

TTibz27 committed:
removed print statements

removed print statements since those will almost certainly cause conflicts

TTibz27 committed:
added exported angle and strength

added an exported angle and velocity variable so people can mess around with those

im-deepfriedwater commented on src/scenes/Baseball-launcher.gd:
### nitpick

generally I've seen script files also follow title casing! so this would be `BaseballLauncher.gd`

im-deepfriedwater commented on src/scenes/Baseball-launcher.gd:
I think the more Godot-like way to structure the baseball launcher would be to create a new scene for the launcher. Probably a Node2D as its root. This also lets us add a timer node as its child which is a little easier to tweak when it shows up in the inspector. Also, it lets us use Godot's inspector to connect to the timer signal instead of having to do it in the script at runtime! It provides a little more highlighting and UI in the inspector to know that this script talks to a signal vs connecting a signal at runtime (though there are times when connecting a signal at runtime is necessary). 
![signal](https://user-images.githubusercontent.com/16827894/226080834-48ff4b59-82ec-41f9-bbcc-ab01e1be7a6e.gif)

im-deepfriedwater replied:
ohhhh I double checked the main scene and I saw you created a Node2D there to add the baseball launcher script, that's probably better since we're not likely to have multiple baseball launchers so there's no need to make a new scene for it. I'd suggest to add a separate timer node as a child of the baseball launcher though!

im-deepfriedwater commented on src/scripts/Baseball.gd:
```diff
@@ -1,10 +1,60 @@
+
+func _physics_process(_delta):
+	#tracks invuln and scaling
+	current_frame +=1
```
I feel like we can use a separate timer here instead of manually counting frames to keep track of invuln to make it easier to work with and tweak!

TTibz27 replied:
hahaha this is both a holdover from me writing my own timers all the time, and also I tried to use the frame count to scale.

I was thinking of doing something like baseball.set_scale(currentframe/40, currentframe/40) so the ball would get progressively bigger, thinking that the set scale function had to be called every frame to maintain it relative to its initial scale. that is absolutely not how it works though, so the ball was almost immediately shrinking down to oblivion lol.

yeah this should definitely be a timer, I think that it would be cool to have the scale adjust progressively but theres probably a lot more graceful ways of handling that lol.

im-deepfriedwater replied:
hahaha ohhh gotcha gotcha, that makes sense!

im-deepfriedwater reviewed (approved):
looks good to me! generally just have nitpick comments on styling and putting timers as child nodes instead of instantiating at runtime!

im-deepfriedwater commented:
also it was very satisfying to just spawn baseballs and watch them fly through the air

TTibz27 committed:
Fixed issues from PR

Renamed a file, repurposed the timer to work for the ball sprite size change, fixed how ball objects get deleted
